plugins/tpm/fu-plugin-tpm.c

```diff
@@ -197,11 +197,63 @@ fu_plugin_tpm_add_security_attr_eventlog(FuPlugin *plugin, FuSecurityAttrs *attr
 	fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_VALID);
 }
 
+static void
+fu_plugin_tpm_add_security_attr_empty(FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FuPluginData *data = fu_plugin_get_data(plugin);
+	gboolean empty;
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GPtrArray) checksums = NULL;
+
+	/* no TPM device */
+	if (data->tpm_device == NULL)
+		return;
+
+	/* create attr */
+	attr = fwupd_security_attr_new(FWUPD_SECURITY_ATTR_ID_TPM_EMPTY_PCR);
+	fwupd_security_attr_set_plugin(attr, fu_plugin_get_name(plugin));
+	fwupd_security_attr_set_level(attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_add_guids(attr, fu_device_get_guids(data->tpm_device));
+	fu_security_attrs_append(attrs, attr);
+
+	if (fu_plugin_has_flag(plugin, FWUPD_PLUGIN_FLAG_DISABLED)) {
+		fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_NOT_FOUND);
+		return;
+	}
```

Comment:
Should this be the first check before checking for no TPM device?

Reply:
It would make sense to me, but I'm pretty sure I copy-pasted this from existing code by @hughsie.

Reply:
Done (and moved in existing code as well).